pkg/sql/logictest/tests/5node-disk/5node-disk_test: TestLogic_distsql_enum failed

pkg/sql/logictest/tests/5node-disk/5node-disk_test.TestLogic_distsql_enum [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8017611?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8017611?buildTab=artifacts#/) on master @ [b6667d9c6ea091303cf02ed2050e0350eada5bcb](https://github.com/cockroachdb/cockroach/commits/b6667d9c6ea091303cf02ed2050e0350eada5bcb):


```
=== RUN   TestLogic_distsql_enum
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/6e9d563e9f7703abdf07517fb35efb2e/logTestLogic_distsql_enum782437834
    test_log_scope.go:79: use -show-logs to present logs inline
[10:28:40] setting distsql_workmem='27818B';
    logic.go:2860: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:68: EXPLAIN (VEC)
        SELECT x from t1 WHERE EXISTS (SELECT x FROM t2 WHERE t1.x=t2.x AND t2.y='hello')
        expected:
            │
            ├ Node 1
            │ └ *colrpc.Outbox
            │   └ *rowexec.joinReader
            │     └ *colfetcher.ColBatchScan
            ├ Node 2
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *rowexec.joinReader
            │   │ └ *colfetcher.ColBatchScan
            │   └ *colrpc.Inbox
            └ Node 3
              └ *colrpc.Outbox
                └ *rowexec.joinReader
                  └ *colfetcher.ColBatchScan
            
        but found (query options: "nodeidx=1,retry") :
            │
            ├ Node 1
            │ └ *colrpc.Outbox
            │   └ *rowexec.joinReader
            │     └ *colfetcher.ColBatchScan
            └ Node 2
              └ *colexec.ParallelUnorderedSynchronizer
                ├ *colrpc.Inbox
                └ *rowexec.joinReader
                  └ *colfetcher.ColBatchScan
[10:29:27] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum: 8 statements
[10:29:27] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum with config 5node-disk: 8 tests, 1 failures
    logic.go:3990: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:88: error while processing
    logic.go:3990: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:88: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6e9d563e9f7703abdf07517fb35efb2e/logTestLogic_distsql_enum782437834
--- FAIL: TestLogic_distsql_enum (49.03s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_distsql_enum.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22590

--- a/pkg/sql/logictest/testdata/logic_test/distsql_enum
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_enum
@@ -44,48 +44,46 @@ ALTER TABLE t2 SPLIT AT VALUES(0),(10),(20);
 ALTER TABLE t1 EXPERIMENTAL_RELOCATE VALUES (ARRAY[1], 0), (ARRAY[2], 10), (ARRAY[3], 20);
 ALTER TABLE t2 EXPERIMENTAL_RELOCATE VALUES (ARRAY[1], 0), (ARRAY[2], 10), (ARRAY[3], 20);
 
-# Tickle stats to force lookup join
-statement ok
-ALTER TABLE t1 INJECT STATISTICS '[
-  {
-    "columns": ["x"],
-    "created_at": "2018-01-01 1:00:00.00000+00:00",
-    "row_count": 10,
-    "distinct_count": 100
-  }
-]'
+query TTTI colnames
+SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t1 WITH DETAILS] ORDER BY lease_holder
+----
+start_key           end_key                 replicas  lease_holder
+<before:/Table/53>  …/1/0                   {1}       1
+…/1/0               …/1/10                  {1}       1
+…/1/10              …/1/20                  {2}       2
+…/1/20              <after:/Table/110/1/0>  {3}       3
 
-statement ok
-ALTER TABLE t2 INJECT STATISTICS '[
-  {
-    "columns": ["x"],
-    "created_at": "2018-01-01 1:00:00.00000+00:00",
-    "row_count": 10000,
-    "distinct_count": 10000
-  }
-]'
+query TTTI colnames
+SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t2 WITH DETAILS] ORDER BY lease_holder
+----
+start_key                 end_key       replicas  lease_holder
+…/1/0                     …/1/10        {1}       1
+…/1/10                    …/1/20        {2}       2
+<before:/Table/109/1/20>  …/1/0         {3}       3
+…/1/20                    <after:/Max>  {3}       3
 
-query T nodeidx=1,retry
+# Ensure that the join readers are planned on the remote nodes.
+query T
 EXPLAIN (VEC)
-SELECT x from t1 WHERE EXISTS (SELECT x FROM t2 WHERE t1.x=t2.x AND t2.y='hello')
+SELECT t1.x from t1 INNER LOOKUP JOIN t2 ON t1.x=t2.x WHERE t2.y='hello'
 ----
 │
 ├ Node 1
+│ └ *colexec.ParallelUnorderedSynchronizer
+│   ├ *rowexec.joinReader
+│   │ └ *colfetcher.ColBatchScan
+│   ├ *colrpc.Inbox
+│   └ *colrpc.Inbox
+├ Node 2
 │ └ *colrpc.Outbox
 │   └ *rowexec.joinReader
 │     └ *colfetcher.ColBatchScan
-├ Node 2
-│ └ *colexec.ParallelUnorderedSynchronizer
-│   ├ *colrpc.Inbox
-│   ├ *rowexec.joinReader
-│   │ └ *colfetcher.ColBatchScan
-│   └ *colrpc.Inbox
 └ Node 3
   └ *colrpc.Outbox
     └ *rowexec.joinReader
       └ *colfetcher.ColBatchScan
 
-query I nodeidx=1
-SELECT x from t1 WHERE EXISTS (SELECT x FROM t2 WHERE t1.x=t2.x AND t2.y='hello')
+query I
+SELECT t1.x from t1 INNER LOOKUP JOIN t2 ON t1.x=t2.x WHERE t2.y='hello'
 ----
 0

